### DISCUSSION TITLE
How to remove all branches created  under a repositories

### DISCUSSION THREAD

**Discussion by ag185262** [community/community#112101] [Repositories]
### Select Topic Area

General

### Body

Hi All,
i have a repository which is old and having lots of branch (more than 500 i believe), and i want to delete all the branches which has not been modified since more than 2 months. 

Please help me if any one knows about it.

Thanks

**Comment by probablynotnmp** [comment]
You can use Git commands and scripting to do so:

Try this code 

```
#!/bin/bash

TWO_MONTHS_AGO=$(date --date="2 months ago" +%Y-%m-%d)

# Get a list of all branches
git fetch --prune
git branch -r --format='%(refname:short)' | while read -r branch; do
    # Get the last commit date for each branch
    last_commit_date=$(git log -1 --format="%ci" "$branch" | awk '{print $1}')

    # Compare last commit date with two months ago
    if [[ "$last_commit_date" < "$TWO_MONTHS_AGO" ]]; then
        # Delete the branch if it's older than two months
        git push origin --delete "$branch"
        echo "Deleted branch: $branch"
    fi
done
```

I haven't tried it myself, I use chatGPT to generate the code but do try it and tell me how it goes.

**Comment by kuciapakrystian** [comment]
scripting with your git client.

#!/bin/bash

# Go to your repository directory
cd /path/to/your/repository

# Fetch all remote branches
git fetch --all

# Iterate over all branches
for branch in $(git branch -r); do
    # Exclude the main branch (change 'main' to your main branch name if it's different)
    if [[ "$branch" != *"origin/main"* ]]; then
        # Check if the branch hasn't been modified in the last 2 months
        if [[ $(git log -n 1 --format="%ct" "$branch") -lt $(date -d '2 months ago' +%s) ]]; then
            # Delete the branch
            git push origin --delete "${branch#*/}"
            echo "Deleted branch: ${branch#*/}"
        fi
    fi
done

Try like this.
